pulp_container/app/viewsets.py

```diff
@@ -545,6 +545,57 @@ class ContainerRepositoryVersionViewSet(RepositoryVersionViewSet):
     """
 
     parent_viewset = ContainerRepositoryViewSet
+    permission_classes = (access_policy.RepositoryVersionAccessPolicyFromDB,)
+
+    DEFAULT_ACCESS_POLICY = {
+        "statements": [
+            {
+                "action": ["list"],
+                "principal": "authenticated",
+                "effect": "allow",
+            },
+            {
+                "action": ["retrieve"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_repo_attr_model_or_obj_perms:container.view_containerrepository",
+            },
+            {
+                "action": ["destroy"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_delete_containerrepository_versions_perms",
+            },
```

Comment:
This is still option one.

Reply:
my understanding is, in case admin would like to change the policy, he will hit this endpoint`http GET  :24817/pulp/api/v3/access_policies` and be able to issue a PUT/PATCH request with changes. If we go with option1. How we would be able to do so? Unless we will change list of `sufficient_repo_permissions` which are hardcoded in the code?
```
$ http GET  :24817/pulp/api/v3/access_policies/92015dd5-a312-49f6-820a-58b91919cce9/
HTTP/1.1 200 OK
Access-Control-Expose-Headers: Correlation-ID
Allow: GET, PUT, PATCH, HEAD, OPTIONS
Connection: close
Content-Length: 2128
Content-Type: application/json
Correlation-ID: 188fd71293a146bc83dd0393f8490a02
Date: Thu, 28 Jan 2021 11:12:14 GMT
Server: gunicorn/20.0.4
Vary: Accept, Cookie
X-Frame-Options: SAMEORIGIN

{
    "permissions_assignment": [
        {
            "function": "add_for_object_creator",
            "parameters": null,
            "permissions": [
                "container.view_containerrepository",
                "container.change_containerrepository",
                "container.delete_containerrepository",
                "container.sync_containerrepository",
                "container.modify_content_containerrepository",
                "container.build_image_containerrepository"
            ]
        }
    ],
    "pulp_created": "2021-01-27T19:15:01.424436Z",
    "pulp_href": "/pulp/api/v3/access_policies/92015dd5-a312-49f6-820a-58b91919cce9/",
    "statements": [
        {
            "action": [
                "list"
            ],
            "effect": "allow",
            "principal": "authenticated"
        },
        {
            "action": [
                "create"
            ],
            "condition": "has_model_perms:container.add_containerrepository",
            "effect": "allow",
            "principal": "authenticated"
        },
        {
            "action": [
                "retrieve"
            ],
            "condition": "has_model_or_obj_perms:container.view_containerrepository",
            "effect": "allow",
            "principal": "authenticated"
        },
        {
            "action": [
                "destroy"
            ],
            "condition": "has_model_or_obj_perms:container.delete_containerrepository",
            "effect": "allow",
            "principal": "authenticated"
        },
        {
            "action": [
                "update",
                "partial_update"
            ],
            "condition": "has_model_or_obj_perms:container.change_containerrepository",
            "effect": "allow",
            "principal": "authenticated"
        },
        {
            "action": [
                "sync"
            ],
            "condition": [
                "has_model_or_obj_perms:container.sync_containerrepository",
                "has_remote_param_model_or_obj_perms:container.view_containerremote"
            ],
            "effect": "allow",
            "principal": "authenticated"
        },
        {
            "action": [
                "remove",
                "tag",
                "untag"
            ],
            "condition": [
                "has_model_or_obj_perms:container.modify_content_containerrepository"
            ],
            "effect": "allow",
            "principal": "authenticated"
        },
        {
            "action": [
                "add"
            ],
            "condition": [
                "has_model_or_obj_perms:container.modify_content_containerrepository",
                "has_content_param_model_or_obj_perms"
            ],
            "effect": "allow",
            "principal": "authenticated"
        },
        {
            "action": [
                "copy_tags",
                "copy_manifests"
            ],
            "condition": [
                "has_model_or_obj_perms:container.modify_content_containerrepository",
                "has_source_param_model_or_obj_perms:container.view_containerrepository"
            ],
            "effect": "allow",
            "principal": "authenticated"
        },
        {
            "action": [
                "build_image"
            ],
            "condition": [
                "has_model_or_obj_perms:container.build_image_containerrepository"
            ],
            "effect": "allow",
            "principal": "authenticated"
        }
    ],
    "viewset_name": "repositories/container/container"
}

```